sonic-xcvrd/xcvrd/sff_mgr.py

```diff
@@ -0,0 +1,407 @@
+        self.helper_logger = helper_logger
+        self.platform_chassis = platform_chassis
+        # port_dict holds data per port entry with logical_port_name as key, it
+        # maintains local copy of the following DB fields:
+        #   CONFIG_DB PORT_TABLE 'index', 'channel', 'admin_status'
+        #   STATE_DB PORT_TABLE 'host_tx_ready'
+        #   STATE_DB TRANSCEIVER_INFO 'type'
+        # plus 'asic_id' from PortChangeEvent.asic_id (asic_id always gets
+        # filled in handle_port_update_event function based on asic_context)
+        # Its port entry will get deleted upon CONFIG_DB PORT_TABLE DEL.
+        # Port entry's 'type' field will get deleted upon STATE_DB TRANSCEIVER_INFO DEL.
+        self.port_dict = {}
+        # port_dict snapshot captured in the previous event update loop
+        self.port_dict_prev = {}
+        self.xcvr_table_helper = XcvrTableHelper(namespaces)
+        self.namespaces = namespaces
+
+    def log_notice(self, message):
+        self.helper_logger.log_notice("SFF: {}".format(message))
+
+    def log_warning(self, message):
+        self.helper_logger.log_warning("SFF: {}".format(message))
+
+    def log_error(self, message):
+        self.helper_logger.log_error("SFF: {}".format(message))
+
+    def on_port_update_event(self, port_change_event):
+        if (port_change_event.event_type
+                not in [port_change_event.PORT_SET, port_change_event.PORT_DEL]):
+            return
+
+        lport = port_change_event.port_name
+        pport = port_change_event.port_index
+        asic_id = port_change_event.asic_id
+
+        # Skip if it's not a physical port
+        if not lport.startswith('Ethernet'):
+            return
+
+        # Skip if the physical index is not available
+        if pport is None:
+            return
+
+        if port_change_event.port_dict is None:
+            return
+
+        if port_change_event.event_type == port_change_event.PORT_SET:
+            if lport not in self.port_dict:
+                self.port_dict[lport] = {}
+            if pport >= 0:
+                self.port_dict[lport]['index'] = pport
+            # This field comes from CONFIG_DB PORT_TABLE. This is the channel
+            # that blongs to this logical port, 0 means all channels. tx_disable
+            # API needs to know which channels to disable/enable for a
+            # particular physical port.
+            if 'channel' in port_change_event.port_dict:
+                self.port_dict[lport]['channel'] = port_change_event.port_dict['channel']
+            # This field comes from STATE_DB PORT_TABLE
+            if 'host_tx_ready' in port_change_event.port_dict:
+                self.port_dict[lport]['host_tx_ready'] = \
+                        port_change_event.port_dict['host_tx_ready']
+            # This field comes from CONFIG_DB PORT_TABLE
+            if 'admin_status' in port_change_event.port_dict and \
```

Comment:
@longhuan-cisco, I think removing the filter on STATE_DB would actually mean that SFF manager is interested in all the fields of STATE_DB and hence, that would be a calculated move.
Agreed that the check is more robust but I think from code perspective, it will be a redundant check since the FILTER option of the STATE_DB is primarily meant to facilitate this.
Also, this might cause unnecessary confusion since other DB field options (such as host_tx_ready) doesn't have DB relevant checks and rather are assumed to be source of truth.I think the case of removing the filter on STATE_DB would actually mean that SFF manager is interested in all the fields of STATE_DB and hence, that would be a calculated move.
Agreed that the check is more robust but I think from code perspective, it will be a redundant check since the FILTER option of the STATE_DB is primarily meant to facilitate this.
Also, this might cause unnecessary confusion since other DB field options (such as host_tx_ready) don't have DB relevant checks and rather are assumed to be source of truth.

We can also add a FILTER for CONFIG_DB so that only the relevant fields are processed by SFF manager

Reply:
Agreed. removed the checks.
And added one comment at the FILTER definition to keep people cautious, since not all people are aware that `admin_status` can also come from STATE_DB.